recordedit/recordEdit.app.js

```diff
@@ -173,7 +171,12 @@
                                 recordEditModel.submissionRows[j] = {};
 
                                 var tuple = page.tuples[j],
-                                values = tuple.values;
+                                    values = tuple.values;
+
+                                // We don't want to mutate the actual tuples associated with the page returned from `reference.read`
+                                // The submission data is copied back to the tuples object before submitted in the PUT request
+                                var shallowTuple = tuple.deepCopyData();
```

Comment:
after addressing the comments in the review of ermrestjs this line ought to read more like
```
var copiedTuple = tuple.copy();
```